### PR TITLE
[clang] Fix direct-initialization with new expressions for arrays

### LINKED ISSUE #78183: [clang] Parenthesized initialization of arrays does not work in new expression
The parenthesized initialization of arrays (aggregates) is not supported in the new expression:
```C++
int main() {
  int x[10](1, 2);  // OK
  auto p = new int[10](1, 2); // ERROR, accepted by GCC and MSVC
}
```
Live example: [Godbolt](https://godbolt.org/z/64K39oEhK)

### PR DESCRIPTION
**Pull request by MitalAshok**
Fixes #78183

Just removed the check entirely. The diagnostic issued for trying to initialize an array (`array initializer must be an initializer list`) is much clearer (than `array 'new' cannot have initialization arguments`).

### CODE REVIEW

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -937,6 +937,9 @@ Bug Fixes to AST Handling
 - Fixed a bug where Template Instantiation failed to handle Lambda Expressions
   with certain types of Attributes.
   (`#76521 <https://github.com/llvm/llvm-project/issues/76521>`_)
+- Fixed a bug where the parenthesized initialization of arrays in a new
+  expression were rejected even in C++20 mode.
```
Comment (cor3ntin) [inline clang/docs/ReleaseNotes.rst:941]:
```suggestion
  expression were incorrectly rejected in C++20 mode.
```

**Review by cor3ntin — APPROVED** [review]
Nice simplification, thanks.
I validated that this does not impact c++98, which just fails in parsing

**Review by shafik — COMMENTED** [review]
I believe the approach here is not sufficient and we have a current PR in progress: https://github.com/llvm/llvm-project/pull/76976

we also need codegen work to implement this.

**Comment by alanzhao1** [comment]
> I believe the approach here is not sufficient and we have a current PR in progress: #76976
> 
> we also need codegen work to implement this.

Correct - here are some examples that will cause this to fail:

```cpp
void foo(int n) {
  new int[n](1, 2);
}
```

```cpp
void bar() {
  new char[]("abcd");
}
```

**Comment by MitalAshok** [comment]
Yeah #76976 supersedes this.

I would still recommend removing `diag::err_new_array_init_args` if only so:

```c++
    int x[2](1, 2);
    int* y = new int[2](1, 2);
```

have similar error messages.

**Pull request closed without merging**